week4-functions/index.html

```diff
@@ -0,0 +1,109 @@
+---
+layout: default
+permalink: /week4-functions/
+---
+    <h2>Quiz 4: Functions</h2>
+    <p>
+      <img src="ice_cream_function.png" width=200 height=200></iframe>
+    </br>
+      1. Recreate the above image by calling <code>function iceCream()</code> in a nested <code>for</code> loop.
+    </p>
+
+<script type="text/p5" data-preview-width="400">
+function setup() { 
+  createCanvas(200, 200);
+  background(255);
+  noStroke();
+  colorMode(HSB, 255);
+  for (________; ________; ________) {
+    for (________; ________; ________) {
+      iceCream(________, ________, ________);
+    }
+  }
+} 
+
+function iceCream(x, y, diameter) {
+  fill(random(360), 112, 331);
+  arc(x, y, diameter, diameter, -PI, 0);
+  fill('#d7c38e');
+  triangle(x - diameter / 2, y + 5, x + diameter / 2, y + 5, x, y + diameter * 1.3);
+}
+</script>
+
+<p>
+  <iframe src="https://alpha.editor.p5js.org/embed/SJDcrKd6" width=250 height=200></iframe>
+  <br/>
+
+  2. Write a program that implements the above sketch. Complete function <code>CtoF()</code> that takes temperature in degree Celsius (°C) as an arguement, and returns temperature in degree Fahrenheit (°F). T(°F) = T(°C) × 1.8 + 32. (Since the p5 widget below doesn't support p5.dom yet, so <code>creatInput</code> won't work, please copy the code below and play with it in other ediors.)</p>
+
+<script type="text/p5" data-preview-width="400">
+function setup() { 
+  createCanvas(200, 200);
+  background(255);
+  var inputTemp = createInput('');
```

Comment:
This is a perfect exercise / question, but we haven't covered even callbacks or any of the p5 DOM stuff.  I would simplify this to make a non-interactive version that just displays the results using `text()`.